File: app/src/main/java/com/odnovolov/randomuser/network/RandomUserService.java
package com.odnovolov.randomuser.network;

import com.odnovolov.randomuser.network.model.RandomUserResponse;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RandomUserService {
    @GET(".")
    Observable<Response<RandomUserResponse>> getRandomUsers(
        @Query("results") int number,
        @Query("gender") String gender
    );
}

File: app/src/main/java/com/odnovolov/randomuser/network/RandomUserService.kt
package com.odnovolov.randomuser.network

import com.odnovolov.randomuser.network.model.RandomUserResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface RandomUserService {
    @GET(".")
    suspend fun getRandomUsers(
        @Query("results") number: Int,
        @Query("gender") gender: String? = null
    ): Response<RandomUserResponse>
}